test/expr/expressions.swift

```diff
@@ -767,6 +767,11 @@ _ = nil == Int.self  // expected-warning {{comparing non-optional value of type
 _ = Int.self != nil  // expected-warning {{comparing non-optional value of type 'any Any.Type' to 'nil' always returns true}}
 _ = nil != Int.self  // expected-warning {{comparing non-optional value of type 'any Any.Type' to 'nil' always returns true}}
 
+_ = Int.self == .none  // expected-warning {{comparing non-optional value of type 'any Any.Type' to 'Optional.none' always returns false}}
+_ = .none == Int.self  // expected-warning {{comparing non-optional value of type 'any Any.Type' to 'Optional.none' always returns false}}
+_ = Int.self != .none  // expected-warning {{comparing non-optional value of type 'any Any.Type' to 'Optional.none' always returns true}}
+_ = .none != Int.self  // expected-warning {{comparing non-optional value of type 'any Any.Type' to 'Optional.none' always returns true}}
+
```

Comment:
The logic seems to handle that case but maybe we should add a test case where value type is an enum with `none` case e.g. 
```swift
enum E {
case none
}
func test(_ e: E) {
_ = e == .none // OK
_ = e == Optional.none // warn
```
Makes sense?

Reply:
Aha yes good point, I'll add a test case for that.